src/solve/form.ts

```diff
@@ -521,17 +503,31 @@ export async function executeRoute(
       );
 
       const tx = await client.signAndBroadcast(msgJSON.sender, [msg], "auto");
+
+      txHash = tx.transactionHash;
     }
 
-    while (true) {
-      console.log("polling...");
+    await skipClient.transaction.track(txHash, multiHopMsg.chain_id);
 
-      const balance = await destinationChainClient.getBalance(
-        destinationChainAddress,
-        denomOut
+    while (true) {
```

Comment:
Is this just gonna be an instant infinite loop? 

Should we have any kind of wait on this?

Reply:
See line 534